### PR TITLE
Deploy LIFI to Worldchain

### PR DESCRIPTION
**Pull request by ezynda3**
# Which Jira task belongs to this PR?

# Why did I implement it this way?

# Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] I have updated any required documentation

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

File: config/networks.json [hunk 1]
```diff
@@ -526,6 +526,23 @@
...
+    "wrappedNativeAddress": "0x4200000000000000000000000000000000000006",
+    "status": "inactive",
+    "type": "mainnet",
+    "rpcUrl": "https://worldchain-mainnet.g.alchemy.com/public",
+    "verificationType": "etherscan",
+    "explorerUrl": "https://worldscan.org",
+    "explorerApiUrl": "https://api.worldscan.org/api",
+    "multicallAddress": "0xcA11bde05977b3631167028862bE2a173976CA11",
+    "safeApiUrl": "https://safe-transaction-worldchain.safe.global/api",
+    "safeAddress": "0xa6E64bC059ddAa72ec374C50B1A5c6b3a27d64Ce",
```
Comment (0xDEnYO) [inline config/networks.json:543]: the newly added property `safeWebUrl` is missing, here an example from Arbitrum:

"safeWebUrl": "https://app.safe.global/home?chain=oeth&safe=arb1:0x9e606d0d2BbA344b911e2F4Eab95d9235A83fe15",

File: script/deploy/safe/add-owners-to-safe.ts [hunk 2]
```diff
@@ -5,7 +5,12 @@ import Safe, { EthersAdapter } from '@safe-global/protocol-kit'
 import SafeApiKit from '@safe-global/api-kit'
 import { ethers } from 'ethers6'
 import { getSafeUtilityContracts, safeAddresses, safeApiUrls } from './config'
-import { getViemChainForNetworkName } from '../../utils/viemScriptHelpers'
+import {
+  NetworksObject,
+  getViemChainForNetworkName,
+} from '../../utils/viemScriptHelpers'
+import data from '../../../config/networks.json'
+const networks: NetworksObject = data as NetworksObject
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/safe/add-owners-to-safe.ts:13]: not needed

File: script/deploy/safe/add-owners-to-safe.ts [hunk 3]
```diff
@@ -40,12 +45,12 @@ const main = defineCommand({
 
     const config: SafeApiKitConfig = {
       chainId: BigInt(chain.id),
-      txServiceUrl: safeApiUrls[network],
+      txServiceUrl: networks[network].safeApiUrl,
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/safe/add-owners-to-safe.ts:48]: Will fail anyway. Not needed

[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/safe/add-owners-to-safe.ts:48]: Will fail anyway

File: script/deploy/safe/add-owners-to-safe.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/safe/add-owners-to-safe.ts:13]: Not needed

File: script/deploy/healthCheck.ts [hunk 4]
```diff
@@ -404,15 +384,13 @@ const main = defineCommand({
       //          │                   SAFE Configuration                    │
       //          ╰─────────────────────────────────────────────────────────╯
       consola.box('Checking SAFE configuration...')
-      if (
-        !globalConfig.safeAddresses[network.toLowerCase()] ||
-        !globalConfig.safeApiUrls[network.toLowerCase()]
-      ) {
+      const networkConfig: Network = networks[network.toLowerCase()]
+      if (!networkConfig.safeAddress || !networkConfig.safeApiUrl) {
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/healthCheck.ts:388]: Will fail anyway

File: script/deploy/_targetState.json [hunk 5]
```diff
@@ -1385,7 +1413,73 @@
...
+        "LiFuelFeeCollector": "1.0.1",
+        "TokenWrapper": "1.0.0",
+        "LiFiDEXAggregator": "1.0.0",
+        "Permit2Proxy": "1.0.0",
+        "AcrossFacetV3": "1.0.0",
+        "AcrossFacetPackedV3": "1.0.0",
+        "ReceiverAcrossV3": "1.0.0",
+        "StargateFacetV2": "1.0.1",
+        "ReceiverStargateV2": "1.0.1"
       }
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/_targetState.json:1483]: This is the latest

File: script/deploy/_targetState.json [hunk 6]
```diff
@@ -1361,6 +1361,34 @@
...
+        "Executor": "2.0.0",
+        "FeeCollector": "1.0.0",
+        "Receiver": "2.0.2",
+        "LiFuelFeeCollector": "1.0.1",
+        "TokenWrapper": "1.0.0",
+        "LiFiDEXAggregator": "1.0.0",
+        "Permit2Proxy": "1.0.0"
+      }
+    }
+  },
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply script/deploy/_targetState.json:1390]: This is ok and expected

File: .env.example [hunk 7]
```diff
@@ -30,6 +30,7 @@ ETH_NODE_URI_ROOTSTOCK=https://rootstock.drpc.org #[pre-commit-checker: not a se
 ETH_NODE_URI_SEI=https://evm-rpc.sei-apis.com #[pre-commit-checker: not a secret].io
 ETH_NODE_URI_SCROLL=https://1rpc.io/scroll #[pre-commit-checker: not a secret].io
 ETH_NODE_URI_TAIKO=https://rpc.taiko.tools #[pre-commit-checker: not a secret].io
+ETH_NODE_URI_WORLDCHAIN=https://worldchain-mainnet.g.alchemy.com/public #[pre-commit-checker: not a secret].io
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply .env.example]: fixed

[the remark this answers is not in the document]
Comment (ezynda3) [reply .env.example]: fixed

File: config/across.json [hunk 8]
```diff
@@ -132,5 +132,13 @@
       "0x5300000000000000000000000000000000000004",
       "0xf55BEC9cafDbE8730f096Aa55dad6D22d44099Df"
     ]
+  },
+  "worldchain": {
+    "chainId": 480,
+    "acrossSpokePool": "0x09aea4b2242abC8bb4BB78D537A67a245A7bEC64",
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply config/across.json:138]: contract addresses are allowed to be the same from chain to chain and often are due to deterministic deployments

File: config/global.json [hunk 9]
```diff
@@ -143,6 +143,7 @@
     "sei": "0x0000000000000000000000000000000000000000",
     "xlayer": "0x0000000000000000000000000000000000000000",
     "taiko": "0x0000000000000000000000000000000000000000",
+    "worldchain": "0x0000000000000000000000000000000000000000",
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply config/global.json:146]: We are trying to deprecate this in favor of `config/networks.json`

File: config/dexs.json [hunk 10]
```diff
@@ -745,6 +745,11 @@
     "0x833Be894C596b15FAe740C2D522d660084c48B05",
     "0x2321F1a63A683a1F3634Dbe1CbA0d657D5F56d54"
   ],
+  "worldchain": [
+    "0x50D5a8aCFAe13Dceb217E9a071F6c6Bd5bDB4155",
+    "0x8f023b4193a6b18C227B4a755f8e28B3D30Ef9a1",
+    "0x603a538477d44064eA5A5d8C345b4Ff6fca1142a"
+  ],
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply config/dexs.json:752]: These are actually used as "DEXs" and are needed to be verified.

File: config/networks.json (hunk 1 above)
Comment (ezynda3) [reply config/networks.json:543]: updated

File: config/networks.json [hunk 11]
```diff
@@ -526,6 +526,23 @@
...
+    "gasZipChainId": 0
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply config/networks.json]: updated

File: deployments/_deployments_log_file.json [hunk 12]
```diff
@@ -22376,6 +22376,20 @@
...
+    "worldchain": {
+      "production": {
+        "1.0.1": [
+          {
+            "ADDRESS": "0xBa713B18c806EcdEEE49FAec623dE2D872192872",
+            "OPTIMIZER_RUNS": "1000000",
+            "TIMESTAMP": "2024-11-19 10:35:41",
+            "CONSTRUCTOR_ARGS": "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "SALT": "",
+            "VERIFIED": "false"
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply deployments/_deployments_log_file.json]: updated

[the remark this answers is not in the document]
Comment (ezynda3) [reply deployments/_deployments_log_file.json]: updated

File: deployments/worldchain.diamond.json [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+                "Name": "GenericSwapFacetV3",
+                "Version": "1.0.0"
+            },
+            "0xe6C6A35684308f2DaadbeeA50B62CFEaAFaa407E": {
+                "Name": "StandardizedCallFacet",
+                "Version": "1.0.0"
+            },
+            "0x1feB868BF64AdC552E051fB7387681F78b988a81": {
+                "Name": "CalldataVerificationFacet",
+                "Version": "1.0.0"
```
Comment (0xDEnYO) [inline deployments/worldchain.diamond.json:50]: Just noticed that we have the outdated version of this facet deployed here....but I guess we can update it when we deploy the new version to all chains...

**Review by 0xDEnYO — DISMISSED** [review]

**Review by 0xDEnYO — APPROVED** [review]

File: deployments/_deployments_log_file.json [hunk 14]
```diff
@@ -22478,6 +22478,20 @@
           }
         ]
       }
+    },
+    "worldchain": {
+      "production": {
+        "1.0.1": [
+          {
+            "ADDRESS": "0xBa713B18c806EcdEEE49FAec623dE2D872192872",
+            "OPTIMIZER_RUNS": "1000000",
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply deployments/_deployments_log_file.json:22487]: nope

File: deployments/_deployments_log_file.json [hunk 15]
```diff
@@ -22478,6 +22478,20 @@
...
+            "TIMESTAMP": "2024-11-19 10:35:41",
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply deployments/_deployments_log_file.json:22488]: not true

File: config/dexs.json [hunk 16]
```diff
@@ -749,6 +749,11 @@
...
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply config/dexs.json:756]: Not a problem

**Pull request merged**